### PR TITLE
TINY-9275: Tooltips

### PR DESCRIPTION
**Pull request by tiny-ben-tran**
Related Ticket:  TINY-9275

Description of Changes:
* New behaviour: `Tooltipping` (the spike PR #8114) 
* Add custom tooltip on various UI: toolbar buttons, dropdowns (no label), icon buttons, color pickers, split buttons and choice buttons
* Tooltip is rendered immediately on focus using keyboard navigation, there will be a slight delay when using mouse to hover
* Good to also review the changes that I've recently made, mainly for reverting the changes that the title attribute was removed, I've added a comment and linked a ticket for places where we need to remove the title attribute.
* https://github.com/tinymce/tinymce/compare/5967e8e458f7783af1e7150255eb4fb3fb2f84c5...3e0198b0c39fe11753b6056be720a3d2ecdb0b6e 	
* The changelog will be updated again, when title attribute is removed and tests are updated in upcoming ticket https://ephocks.atlassian.net/browse/TINY-10453

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [ ] Docs ticket created (if applicable)

GitHub issues (if applicable):

### CODE REVIEW

File: modules/alloy/src/main/ts/ephox/alloy/behaviour/tooltipping/ActiveTooltipping.ts [hunk 1]
```diff
@@ -62,12 +62,22 @@ const events = (tooltipConfig: TooltippingConfig, state: TooltippingState): Allo
...
-        }, tooltipConfig.delay);
+        }, tooltipConfig.delayForShow());
       }),
       AlloyEvents.run(HideTooltipEvent, (comp) => {
         state.resetTimer(() => {
           hide(comp);
-        }, tooltipConfig.delay);
+        }, tooltipConfig.delayForHide());
+      }),
+      AlloyEvents.run(ImmediateShowTooltipEvent, (comp) => {
```
Comment (tiny-ben-tran) [inline modules/alloy/src/main/ts/ephox/alloy/behaviour/tooltipping/ActiveTooltipping.ts:72]: Instant tooltip renders for keyboard users

File: modules/tinymce/src/themes/silver/main/ts/backstage/TooltipsBackstage.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+export interface TooltipsProvider {
+  readonly getConfig: (spec: { tooltipText: string }) => TooltippingTypes.TooltippingConfigSpec;
+}
+
+export const TooltipsBackstage = (
+  getSink: () => Result<AlloyComponent, any>
+): TooltipsProvider => {
+
+  const tooltipDelay = 800;
+  const intervalDelay = tooltipDelay * 0.2;   // Arbitrary value
```
Comment (tiny-ben-tran) [inline modules/tinymce/src/themes/silver/main/ts/backstage/TooltipsBackstage.ts:13]: Might do trials and errors to find a delay rate that will give a better UX

File: modules/tinymce/src/themes/silver/main/ts/ui/general/Tooltips.ts [hunk 3]
```diff
@@ -1,16 +1,19 @@
...
 
 // TODO: Consider using this once we fix the delayed attempt to appear after it's gone problem.
 
+// FIX this with a proper design / architecture.
+let numActiveTooltips = 0;
+
 // TODO: Make the arrow configurable.
-const upConfig = (item: { text: string }, sharedBackstage: UiFactoryBackstageShared): TooltippingBehaviour => Tooltipping.config({
-  delay: 200,
+const upConfig = (item: { text: string }, getSink: () => Result<AlloyComponent, any>): TooltippingBehaviour => Tooltipping.config({
```
Comment (tiny-ben-tran) [inline modules/tinymce/src/themes/silver/main/ts/ui/general/Tooltips.ts:12]: - [ ] Working toward deleting this file

File: modules/tinymce/src/themes/silver/main/ts/backstage/TooltipsBackstage.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    /*
+      <div class="tox-tooltip tox-tooltip--down">
+  <div class="tox-tooltip__body">Are you hanging on the edge of your seat? </div>
+  <i class="tox-tooltip__arrow"></i>
+  </div>*/
+    tooltipDom: {
+      tag: 'div',
+      classes: [ 'tox-tooltip', 'tox-tooltip--up' ]
+    },
+    tooltipComponents: [
```
Comment (tiny-ben-tran) [inline modules/tinymce/src/themes/silver/main/ts/backstage/TooltipsBackstage.ts:33]: Allow customization of tooltips?

File: modules/tinymce/src/themes/silver/main/ts/ui/general/Tooltips.ts (hunk 3 above)
Comment (shanmen-tiny) [reply modules/tinymce/src/themes/silver/main/ts/ui/general/Tooltips.ts]: File has been removed.

File: modules/tinymce/src/themes/silver/main/ts/backstage/TooltipsBackstage.ts (hunk 2 above)
Comment (shanmen-tiny) [reply modules/tinymce/src/themes/silver/main/ts/backstage/TooltipsBackstage.ts:13]: This will be addressed in an upcoming ticket. TINY-10475

**Pull request closed without merging**